### PR TITLE
Fix current tree path preselected folder on MediaGallery, after reopen instance

### LINKED ISSUE #29310: Requested default folder is not selected on opening media gallery (current_tree_path parameter)
### Preconditions (*)
1. Install Magento 2.4.x

### Steps to reproduce (*)
1. Open media gallery from the category image field
2. Open media gallery from the WYSIWYG

### Expected result (*)
If no filters were applied (except directory) the default directory should be selected based on `current_tree_path` parameter
1. catalog/category directory is selected by default
2. WYSIWYG directory is selected by default

### Actual result (*)
The previously selected directory is selected by default

### Additional details
The directory that should be selected by default is provided via `current_tree_path` parameter

### PR DESCRIPTION
**Pull request by Nazar65**
### Related Pull Requests
https://github.com/magento/magento2-page-builder/pull/632

### Fixed Issues (if relevant)

1. Fixes https://github.com/magento/magento2/issues/29310 Requested default folder is not selected on opening media gallery (current_tree_path parameter)
### Manual testing scenarios (*)
1. Open media gallery from the category image field
2. Open media gallery from the WYSIWYG

### Expected result (*)
1. catalog/category directory is selected by default
2. WYSIWYG directory is selected by default

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Cms/view/adminhtml/web/js/folder-tree.js [hunk 1]
```diff
@@ -78,30 +79,92 @@ define([
...
-                };
+                });
+                paths.unshift('root');
+                paths.reverse();
+
+                return paths;
+            };
+
+            $(window).on('reload.MediaGallery', function () {
+                pathId =  window.MediabrowserUtility.pathId.replace(',,', '--');
```
Comment (gabrieldagama) [inline app/code/Magento/Cms/view/adminhtml/web/js/folder-tree.js]: From our investigation it looks like the `,,` on the pathId (which comes from current_tree_path) just exists when we open the gallery from the uploader button, follow the request example:
`https://magento.local/admin/cms/wysiwyg_images/index/target_element_id/E5U0XX4/store/0/type/image/?isAjax=true&current_tree_path=Y2F0YWxvZy9jYXRlZ29yeQ,,`

But when we open from wysiwyg the problem isn't there:

`https://magento.local/admin/cms/wysiwyg_images/index/current_tree_path/d3lzaXd5Zw--/target_element_id/category_form_description/store/0/type/image/?isAjax=true`

It is probably related to the way the url parameters is being set, can we review the uploader button and fix it to not replace the `-` character? By doing that this replace can be removed.

File: app/code/Magento/Cms/view/adminhtml/web/js/folder-tree.js [hunk 2]
```diff
@@ -78,30 +79,92 @@ define([
...
+                    decodedPath = _decodePathId(window.MediabrowserUtility.pathId.replace(/--|,,/, '==')).split('/');
 
-                    lastExistentFolderEl = folderEl;
+                $.each(decodedPath, function (i, val) {
+                    var isLastElement = i === decodedPath.length - 1;
 
-                    if (path.length) {
-                        tree.jstree('open_node', folderEl, recursiveOpen);
+                    if (isLastElement) {
+                        paths[i] = window.MediabrowserUtility.pathId.replace(',,', '--');
```
Comment (gabrieldagama) [inline app/code/Magento/Cms/view/adminhtml/web/js/folder-tree.js]: By fixing the uploader button url we will be able to remove this replace as well.

File: app/code/Magento/Cms/view/adminhtml/web/js/folder-tree.js [hunk 3]
```diff
@@ -78,30 +79,92 @@ define([
...
+                    .replace(/\//g, '_')
+                    .replace(/=/g, '-');
+            },
+
+            /**
+             * Decode path value
+             *
+             * @param {String} val
+             */
+            _decodePathId = function (val) {
```
Comment (gabrieldagama) [inline app/code/Magento/Cms/view/adminhtml/web/js/folder-tree.js]: Here we should replace the characters first and decode later. Exactly the opposite that the encodePathId is doing.

**Review by gabrieldagama — CHANGES_REQUESTED** [review]

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@mzgento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
looks like the "page builder" tests affected, need to recheck failed tests

**Comment by gabrieldagama** [comment]
The risk was set to `low` due to: the changes applied to this PR affect just a specific area on the admin.

**Comment by engcom-Golf** [comment]
@magento run all tests

**Comment by sidolov** [comment]
@Nazar65  I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Pull request closed without merging**